src/transforms/reduce/transform.rs

```diff
@@ -760,4 +754,73 @@ merge_strategies.bar = "concat"
         })
         .await;
     }
+
+    #[tokio::test]
+    async fn nested_path_strategy() {
+        let reduce_config = toml::from_str::<ReduceConfig>(indoc!(
+            r#"
+            group_by = [ "g" ]
+
+            merge_strategies.g = "discard"
+            merge_strategies."a.b.c" = "array"
+
+            [ends_when]
+              type = "vrl"
+              source = "exists(.test_end)"
+            "#,
+        ))
+        .unwrap();
+
+        assert_transform_compliance(async move {
+            let (tx, rx) = mpsc::channel(1);
+
+            let new_schema_definition = reduce_config.outputs(
+                TableRegistry::default(),
+                &[(OutputId::from("in"), Definition::default_legacy_namespace())],
+                LogNamespace::Legacy,
+            )[0]
+            .clone()
+            .log_schema_definitions
+            .get(&OutputId::from("in"))
+            .unwrap()
+            .clone();
+
+            let (topology, mut out) = create_topology(ReceiverStream::new(rx), reduce_config).await;
+
+            let mut e_1 = LogEvent::from("foo");
+            e_1.insert("a.b.c", json!(1));
+            e_1.insert("g", json!(777));
+            let mut metadata_1 = e_1.metadata().clone();
+            metadata_1.set_upstream_id(Arc::new(OutputId::from("reduce")));
+            metadata_1.set_schema_definition(&Arc::new(new_schema_definition.clone()));
+
+            tx.send(e_1.into()).await.unwrap();
+
+            let mut e_2 = LogEvent::from("bar");
+            e_2.insert("a.b.c", 2);
+            e_2.insert("g", json!(777));
+            e_2.insert("test_end", "done");
+            tx.send(e_2.into()).await.unwrap();
+
+            let mut output = out.recv().await.unwrap().into_log();
+            output.remove_timestamp();
+            output.remove("timestamp_end");
+
+            assert_eq!(
+                *output.value(),
+                btreemap! {
+                    "message" => "foo",
+                    "a.b.c" => json!([1, 2]),
```

Comment:
I don't think we are testing here nested fields - we are testing a field {"a.b.c" = "strategy", not {"a": "b": "c": "strategy"}. It should be:
```
"a" => btreemap!{ "b" => btreemap!{"c" => json!([1, 2])}},
```
which means that we still do not support nested fields?